### DISCUSSION TITLE
How does hydration work?

### DISCUSSION THREAD

**Discussion by dbilgin** [TanStack/query#1312] [Q&A]
How do we actually use [hydration](https://react-query.tanstack.com/docs/api#hydrationdehydrate)? I've been trying to make it work in my project, but no chance. Are there any examples for this?

I dehydrate the state like this

```
const dehydratedState = dehydrate(queryCache, {
    shouldDehydrate: () => true,
});
```
But then I'm lost on hydration as it takes in the dehydrated state. I tried below, but it doesn't work:
```
const dehydratedState = dehydrate(queryCache, {
    shouldDehydrate: () => true,
});
hydrate(dehydratedState);
```

**Comment by cjpdee** [comment]
I am also lost on this, I can't suss out how to use hydration from the docs. I'd like to put my stale prefetched state from getStaticProps (using  ISR) onto my page and refetch once the page has loaded.
